app/controllers/citizens/identify_types_of_incomes_controller.rb

```diff
@@ -25,13 +26,16 @@ def transaction_types
     end
 
     def none_selected?
-      params[:none_selected] == 'true' && remove_existing_transaction_types
+      params[:none_selected] == 'true' &&
+        remove_existing_transaction_types &&
+        legal_aid_application.update!(no_credit_transaction_types_selected: true)
     end
 
     def transactions_added
       return if transaction_types.empty?
 
       remove_existing_transaction_types
+      legal_aid_application.update!(no_credit_transaction_types_selected: false)
```

Comment:
Is it worth putting the two actions into a transaction? Otherwise if the application currently has no transaction types and `legal_aid_application.transaction_types << transaction_types` fails, `no_credit_transaction_types_selected` will still be set as `false`